### ISSUE TITLE
Yarn RW Prisma migrate Dev Breaks for Database_URL from Supabase

### ISSUE THREAD

**Issue by ngoctranfire** [redwoodjs/redwood#5767]
I'm trying to start a Redwood Project and I want to use Supabase Postgresql Database. However, I can't seem to connect and run migrations on my Database. I'd love to get some help with this.

Here's my error:

```
Environment variables loaded from .env
Prisma schema loaded from api/db/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.mydatabase.supabase.co:6543"

Error: db error: FATAL: unexpected response from login query
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:250
             
```

**Comment by jtoar** [comment]
Hey @ngoctranfire sorry we never got back to you about this! There's quite a few Redwood apps out there using Supabase, so it should be more than possible to do so. Did you end up getting it working? And when you got this error, was it while following the docs here? https://supabase.com/docs/guides/with-redwoodjs

**Comment by taivo** [comment]
I ran into this issue recently too and stumbled upon the need to create a shadow database (for prisma to perform diffs during migration). You can get more info here https://supabase.com/docs/guides/integrations/prisma#configuring-the-project-to-use-postgresql

TLDR
1. Use a psql client to connect to your supabase db as the admin `postgres` user and create an empty database (example name `postgres_shadow`)
2. In your .env file, add a `SHADOW_DATABASE_URL` variable whose value is the same as `DATABASE_URL` except the database name (the last part) would be from the db created in step 1.
3. In `api/db/schema.prisma`, add the shadow db to your data source config. It looks something like this 
```
datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
}
```

**Comment by jtoar** [comment]
Thanks @taivo! I'll reference it from https://github.com/redwoodjs/redwood/issues/2091 since it's related to that issue.

**Comment by Jolg42** [comment]
@jtoar In the issue description the port is the `pgbouncer` one, `6543` which causes issues with Migrate.

There is a Prisma issue related here https://github.com/prisma/prisma/issues/14069

You can always use the non connection pooled connection from Supabase for migrations to make sure to avoid this problem.

In the next Prisma release, `4.10.0` you will be able to add a `directUrl` field to the datasource block.

**Comment by jtoar** [comment]
Hey @Jolg42 thanks for for pointing me to that issue, I'll save it for future reference. And `directUrl` sounds really nice, we'll be sure to highlight that in our docs when Prisma `v4.10.0` is out.
